format/hostname.go

```diff
@@ -52,19 +52,31 @@ func (format *Hostname) Configure(conf core.PluginConfigReader) error {
 
 // ApplyFormatter update message payload
 func (format *Hostname) ApplyFormatter(msg *core.Message) error {
+	content := format.getFinalContent(format.GetAppliedContent(msg))
+	format.SetAppliedContent(msg, content)
+
+	return nil
+}
+
+func (format *Hostname) getFinalContent(content []byte) []byte {
 	hostname, err := os.Hostname()
 	if err != nil {
 		format.Log.Error.Print(err)
 		hostname = "unknown host"
 	}
 
-	dataSize := len(hostname) + len(format.separator) + msg.Len()
-	payload := core.MessageDataPool.Get(dataSize)
+	lenSeparator := len(format.separator)
+	if lenSeparator > 0 {
```

Comment:
Other formatters follow this pattern, too.
It hink in case of no separator we should do the logical thing - don't use a separator.
Storing just the separator is a totally different thing and can be achieved by using the clear formatter beforehand.